tensorflow/g3doc/hardware/adding_support/fake_device.md

```diff
@@ -0,0 +1,278 @@
+# Constructing a Fake Device in TensorFlow
+
+A good starting point to adding new hardware is to add 
+existing hardware under a different name! So here we add
+the standard CPU under the name SPU by constructing a 
+set of classes called ThreadPiscine (piscine means pool
+in French).
+
+We will create a new ThreadPool under a different name in 
+the C++ code. Throughout this section, I will refer to it 
+as ThreadPiscine (ie. pool -> piscine) and the hardware as
+ a SPU (ie. C -> S). I will reimplement the code from 
+TensorFlow with these names.
+
+The diff for the minimal changes is broken up into parts here
+
+```bash
```

Comment:
As discussed in the comments, this shouldn't really be necessary -- we need to add the APIs to TensorFlow to dynamically add a new device without requiring any source code modifications, so we should probably elide this for now.